### ISSUE TITLE
Validate and update traject for name_search request handler

### ISSUE THREAD

**Issue by mejackreed** [projectblacklight/arclight#656]
This aims to gain parity. Currently we can do comparisons on stored fields but its more difficult to compare indexed only fields. Make sure that we are indexing also the non-stored fields.

`name_search` seems to be searching these fields:
https://github.com/projectblacklight/arclight/blob/master/solr/conf/solrconfig.xml#L202-L210

Failing test:

```
  11) Highlighted search results when querying #name highlights the snippets
      Failure/Error: expect(page).to have_css 'em', text: 'William', count: 3
        expected to find visible css "em" with text "William" within #<Capybara::Node::Element tag="div" path="/html/body/div[1]/div[6]/section[1]/div[2]/article/div[7]"> 3 times, found 2 matches: "William", "William". Also found "Root", "Root", which matched the selector but not all filters.
      # ./spec/features/highlighted_search_results_spec.rb:36:in `block (6 levels) in <top (required)>'
      # ./spec/features/highlighted_search_results_spec.rb:35:in `block (5 levels) in <top (required)>'
      # ./spec/features/highlighted_search_results_spec.rb:34:in `block (4 levels) in <top (required)>'
```

**Comment by mejackreed** [comment]
Upon further review, I'm not sure the original ticket approach was exactly correct. The highlight fields to not appear to be populating 100% correctly. Looking into this further.

**Comment by mejackreed** [comment]
Resolved in #666
